scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        -----
+        This function calculates the differential entropy using the natural
+        logarithm; i.e. the logarithm with base :math:`e`. Consequently, the
+        value is expressed in (dimensionless) "units" of nats. To convert the
+        entropy to different units (i.e. corresponding with a different base),
+        divide the result by the natural logarithm of the desired base.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1., b=1.)
+
+        Evaluate the differential entropy:
+
+        >>> X.entropy()
+        0.6931471805599454
+
+        """
+        return self._entropy_dispatch(method=method, **self._parameters)
+
+    @_dispatch
+    def _entropy_dispatch(self, method=None, **kwargs):
+        if self._overrides('_entropy_formula'):
+            method = self._entropy_formula
+        elif self._overrides('_logentropy_formula'):
+            method = self._entropy_logexp
+        else:
+            method = self._entropy_quadrature
+        return method
+
+    def _entropy_formula(self, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _entropy_logexp(self, **kwargs):
+        return np.real(np.exp(self._logentropy_dispatch(**kwargs)))
+
+    def _entropy_quadrature(self, **kwargs):
+        def integrand(x, **kwargs):
+            pdf = self._pdf_dispatch(x, **kwargs)
+            return np.log(pdf)*pdf
+        return -self._quadrature(integrand, kwargs=kwargs)
+
+    @_set_invalid_nan_property
+    def median(self, *, method=None):
+        r"""Median
+
+        If a continuous random variable :math:`X` has probability :math:`0.5` of
+        taking on a value less than :math:`m`, then :math:`m` is the median.
+        That is, the median is the value :math:`m` for which:
+
+        .. math::
+
+            P(X ≤ m) = 0.5 = P(X ≥ m)
+
+        Parameters
+        ----------
+        method : {None, 'formula', 'icdf'}
+            The strategy used to evaluate the median.
+            By default (``None``), the infrastructure chooses between the
+            following options.
+
+            - ``'formula'``: use a formula for the median
+            - ``'icdf'``: evaluate the inverse CDF of 0.5
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The median
+
+        See Also
+        --------
+        ContinuousDistribution.mean
+        ContinuousDistribution.mode
+        ContinuousDistribution.icdf
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=0., b=10.)
+
+        Evaluate the inverse CDF at the desired argument:
```

Comment:
rather: compute the median and compare it to the icdf/iicdf

Reply:
Right. Left it at "Compute the median:" like every other one of these examples where I compare against a reference.

Reply:
can be resolved